packages/cli/src/lib/utils.test.ts

```diff
@@ -73,25 +75,33 @@ describe('utils', () => {
   });
 
   describe('getAxeSource', () => {
-    describe('mock file', () => {
-      beforeEach(() => {
-        mock({
-          '/node_modules/axe-core': {},
-          '../node_modules/axe-core': {
-            'axe.js': mock.load(require.resolve('axe-core'))
-          }
-        });
+    describe.only('mock file', () => {
+      let dirname = '<NOT A FILE>';
+      before(() => {
+        const axeVersionCheck = dependencies['axe-core'].replace('^', '');
+        const tempDir = tempy.directory();
+        mkdirSync(join(tempDir, 'node_modules'));
+        mkdirSync(join(tempDir, 'node_modules', 'axe-core'));
+        writeFileSync(
+          join(tempDir, 'node_modules', 'axe-core', 'axe.js'),
+          `"${axeVersionCheck}"`
+        );
+
+        mkdirSync(join(tempDir, 'packages'));
+        mkdirSync(join(tempDir, 'packages', 'cli'));
+        mkdirSync(join(tempDir, 'packages', 'cli', 'node_modules'));
+        mkdirSync(join(tempDir, 'packages', 'cli', 'node_modules', 'axe-core'));
+        dirname = join(tempDir, 'packages', 'cli', 'lib');
+        mkdirSync(dirname);
       });
 
-      afterEach(() => {
-        mock.restore();
-      });
       it('fall back to use `locally` installed axe-core', () => {
-        const axeSource = utils.getAxeSource();
+        const axeSource = utils.getAxeSource(undefined, dirname, dirname);
```

Comment:
With this update, I feel a bit confused about what this test is trying to actually verify - I'm not sure there's any meaningful real case where `cwd` === `dirname` in practice. I think it would be preferable to break this out into separate test cases for:

* `it('uses axe.js from the working directory if it exists')` (where test setup places `axe.js` in all 3 locations with different version numbers, to verify which version is actually found)
* `it("falls back to axe-core from the working directory's node_modules if axe.js doesn't exist in the working directory")` (where test setup places `axe.js` in both the dirname and cwd based node_modules paths, but not in cwd, again with different versions to verify which one is found)
* `it("falls back to axe-core from our own package's node_modules if no working-directory based implementation exists")` (where test setup only populates the dirname based node_modules path)